src/main/java/org/spongepowered/common/mixin/core/entity/item/EntityItemMixin.java

```diff
@@ -192,6 +192,21 @@
             ci.cancel();
         }
     }
+    
+    @Inject(
+        method = "onUpdate", 
+        at = @At (
+            value = "INVOKE", 
+            ordinal = 1, 
+            target = "Lnet/minecraft/entity/item/EntityItem;setDead()V")
+    )
+    private void spongeImpl$fireExpireEntityEventTargetItem(final CallbackInfo ci) {
+        try (CauseStackManager.StackFrame frame = Sponge.getCauseStackManager().pushCauseFrame()) {
```

Comment:
Is it guaranteed that we're on the main thread in single player? I remember at times there's a lot that is written assuming this is run on the server only to blow up on the client.

Reply:
I have no idea how to verify this without running the jar on my client, and I cannot seem to get my compiled jar to work on the client. As noted above I'm very inexperienced with this stuff.